src/sentry/ingest/transaction_clusterer/tasks.py

```diff
@@ -86,6 +90,7 @@ def cluster_projects(projects: Sequence[Project]) -> None:
                 # Clear transaction names to prevent the set from picking up
                 # noise over a long time range.
                 redis.clear_samples(ClustererNamespace.TRANSACTIONS, project)
+            pending.remove(project)
```

Comment:
My intuition tells me the unclustered projects should be batched together at the end of the input sequence, but I don't want to make that assumption and I prefer keeping track of projects separately. We could also log the indexes in the input sequence, but I don't see how that could be useful.